src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,402 @@
+
+    public static final String LINE = "____________________________________________________________";
+
+    public static void writeToDisk(ArrayList<Task> tasks) throws IOException{
+        //Each time I want to write to the Disk, I clear away the entire contents of the data file and write everything
+        //Done by deleting the file and creating it again
+        File clear = new File(String.valueOf(dataFileDirectory));
+        clear.delete();
+        createNewFile();
+
+        //To write to the new file, I will append 1 task from the ArrayList at a time
+        FileWriter fw = new FileWriter(String.valueOf(dataFileDirectory), true);
+        for(Task task : tasks){
+            String temp = "";
+
+            if(task.getDoneStatus()){
+                temp += "1|";
+            } else {
+                temp += "0|";
+            }
+            temp += task.getType() + "|" + task.getDescription() + "|" + task.getSeparator() + "|";
+            if(task instanceof Event) {
+                temp += ((Event) task).getAt();
+            } else if(task instanceof Deadline){
+                temp += ((Deadline) task).getBy();
+            } else {
+                temp += "";
+            }
+            temp += "\n";
+            fw.write(temp);
+        }
+        fw.close();
+    }
+
+    public static void wrapWriteToDisk(ArrayList<Task> tasks){
+        try{
+            writeToDisk(tasks);
+        } catch (IOException e) {
+            System.out.println("Issues with writing to .txt file");
+        }
+    }
+
+    public static void loadDataFromDisk(ArrayList<Task> tasks) throws FileNotFoundException {
+        File dataFile = new File(String.valueOf(dataFileDirectory));
+        Scanner s = new Scanner(dataFile);
+
+        while(s.hasNext()){
+            //format I am using is, to use index 0 to store the status.
+            //E.g:
+            //1|event|memes   r|/at|   next Friday
+            String temp = s.nextLine();
+            String[] command = temp.split("\\|");
+            Task tempTask = populateArrayList(Arrays.copyOfRange(command,1, command.length), tasks);
+            if (command[0].equals("1")){
+                tempTask.setAsDone();
+            }
+        }
+        s.close();
+    }
+
+    public static void createNewFile() {
+        try{
+            // If directory does not exist, then create the directory
+            // If directory exists already, just does nothing
+            checkForDirectory();
+        } catch (IOException e) {
+            System.out.println("A serious error has occurred");
+        }
+
+        try{
+            // If file does not exist, then create the file
+            checkForFile();
+        } catch (IOException e) {
+            System.out.println("A serious error has occurred");
+        }
+    }
+
+    public static void checkForFile() throws IOException {
```

Comment:
The name of the method is "checkForFile", but it does more than just check: It also creates new file if there is no file in the given path.